src/twister2/quarantine_plugin.py

```diff
@@ -95,37 +100,51 @@ def get_matched_quarantine(self, item: pytest.Item) -> QuarantineElement | None:
         """Return quarantine element if test is matched to quarantine rules"""
         scenario = item.originalname
         platform = get_item_platform(item)
-        architecture = item.config.twister_config.get_platform(platform).arch if platform else ''
+        if platform:
+            plat_spec = item.config.twister_config.get_platform(platform)
+            architecture = plat_spec.arch
+            simulation = plat_spec.simulation
+        else:
+            architecture = ''
+            simulation = ''
 
         for qelem in self.qlist:
             matched: bool = False
             if qelem.scenarios:
-                if scenario in qelem.scenarios:
-                    matched = True
-                else:
-                    matched = False
+                if (matched := _is_element_matched(scenario, qelem.scenarios, qelem.use_regex)) is False:
                     continue
             if qelem.platforms:
-                if platform in qelem.platforms:
-                    matched = True
-                else:
-                    matched = False
+                if (matched := _is_element_matched(platform, qelem.platforms, qelem.use_regex)) is False:
                     continue
             if qelem.architectures:
-                if architecture in qelem.architectures:
-                    matched = True
-                else:
-                    matched = False
+                if (matched := _is_element_matched(architecture, qelem.architectures, qelem.use_regex)) is False:
                     continue
+            if qelem.simulations:
+                if (matched := _is_element_matched(simulation, qelem.simulations, qelem.use_regex)) is False:
+                    continue
+
             if matched:
                 return qelem
-
         return None
 
 
+def _is_element_matched(element: str, list_of_elements: list, use_regex=False) -> bool:
+    """Return True if given element is matching to any of elements from the list"""
+    if use_regex:
```

Comment:
could be written with this way (less indentions, less complexity):
```python
if not user_regex:
    if element in list_of_elements:
        return True
for pattern in list_of_elements:
    if re.fullmatch(pattern, element):
        return True			
```

Reply:
changed, thanks!